### PR TITLE
Fix typo in code sample

### LINKED ISSUE #154: Documentation Suggestion - GetReadOnlySequence readme.md example incorrect
The example code for GetReadOnlySequence in the root readme.md appears to be incorrect:

```csharp
using (var stream = manager.GetStream())
using (var sha256Hasher = IncrementalHash.CreateHash(HashAlgorithmName.SHA256))
{
    foreach (var memory in stream.GetReadOnlySequence()) // Stream is a MemoryStream NOT a RecyclableMemoryStream
    {
        sha256Hasher.AppendData(memory.Span);
    }
....
```
The stream is an MemoryStream and so we'd need to cast as a RecyleableMemoryStream for the call to `GetReadonlySequence()`.  Or am I missing something?

### PR DESCRIPTION
**Pull request by benmwatson**
Resolves #154

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -173,7 +173,7 @@ using (var stream = manager.GetStream())
 `GetReadOnlySequence` returns a [ReadOnlySequence<byte>](https://docs.microsoft.com/en-us/dotnet/api/system.buffers.readonlysequence-1?view=netstandard-2.1) that can be used for zero-copy stream processing. For example, hashing the contents of a stream: 
 
 ```
-using (var stream = manager.GetStream())
+using (var stream = manager.GetStream() as RecyclableMemoryStream)
```
Comment (grbell-ms) [inline README.md:176]: The code on line 163 has the same issue.

**Review by grbell-ms — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (benmwatson) [reply README.md:176]: Thanks, good catch.

**Review by paulomorgado — APPROVED** [review]
Maybe we should change `GetStream` to return `RecyclableMemoryStream` instead.

There's no point in returing `MemoryStream`.

**Comment by grbell-ms** [comment]
> Maybe we should change `GetStream` to return `RecyclableMemoryStream` instead.
> 
> There's no point in returing `MemoryStream`.

I believe the change would maintain binary compatibility.

**Comment by benmwatson** [comment]
> > Maybe we should change `GetStream` to return `RecyclableMemoryStream` instead.
> > There's no point in returing `MemoryStream`.
> 
> I believe the change would maintain binary compatibility, so we should definitely return RecyclableMemoryStream.

Definitely worth considering. I'll add an issue to track this.

**Pull request merged**